waveform_editor/shape_editor/coil_currents.py

```diff
@@ -0,0 +1,143 @@
+            coil_current = coil.current
+            checkbox = pn.widgets.Checkbox(value=False, margin=(30, 10, 10, 10))
+            slider = pn.widgets.EditableFloatSlider(
+                name=f"{coil.name}",
+                value=coil_current.data[0],
+                start=-5e4,
+                end=5e4,
+                disabled=True,
+                format=PrintfTickFormatter(
+                    format=f"%.0f {coil_current.metadata.units}"
+                ),
+                width=450,
+            )
+
+            def toggle_slider(event, slider=slider):
+                slider.disabled = not event.new
+
+            checkbox.param.watch(toggle_slider, "value")
+            row = pn.Row(checkbox, slider)
+            new_coil_ui.append(row)
+
+        self.coil_ui = new_coil_ui
+
+    def fill_pf_active(self, pf_active):
+        """Update the coil currents of the provided pf_active IDS. Only coils with
+        their corresponding checkbox checked are updated.
+
+        Args:
+            pf_active: pf_active IDS to update the coil currents for.
+        """
+        for i, coil_ui in enumerate(self.coil_ui):
+            checkbox, slider = coil_ui.objects
+            if checkbox.value:
+                pf_active.coil[i].current.data = np.array([slider.value])
+
+    def sync_ui_with_pf_active(self, pf_active):
+        """Synchronize UI sliders with the current values from the pf_active IDS.
+
+        Args:
+            pf_active: pf_active IDS for which the coil currents are used.
+        """
+        for i, coil in enumerate(pf_active.coil):
+            _, slider = self.coil_ui[i].objects
+            slider.value = coil.current.data[0]
+
+    def update_fixed_coils_in_xml(self, xml_string):
+        """Generate XML parameters indicating which coils are fixed based on
+        UI checkboxes.
+
+        Args:
+            xml_string: XML string representing configuration parameters.
+
+        Returns:
+            Updated XML string with the `n_group_fixed_index` and `group_fixed_index`
+                elements modified.
+        """
+        root = ET.fromstring(xml_string)
+        coil_groups = list(map(int, root.find("coil_group_index").text.split()))
+
+        fixed_coils = [i for i, row in enumerate(self.coil_ui) if row.objects[0].value]
+
+        target_groups = {coil_groups[coil_idx] for coil_idx in fixed_coils}
+        fixed_groups = sorted(list(target_groups))
+
+        n_group_fixed_index = len(fixed_groups)
+        if n_group_fixed_index == 0:
+            # NICE requires group_fixed_index to be filled even when there are no fixed
+            # coils, so set to invalid index
+            group_fixed_index = "-1"
+        else:
+            group_fixed_index = " ".join(map(str, fixed_groups))
+
+        params = {
+            "n_group_fixed_index": n_group_fixed_index,
+            "group_fixed_index": group_fixed_index,
+        }
+
+        return self._update_xml_params(root, params)
```

Comment:
This seems a bit overcomplicated, why not directly update the XML here and get rid of the `_update_xml_params` method?

```suggestion
        root.find("n_group_fixed_index").text = str(len(fixed_groups))
        # NICE requires group_fixed_index to be filled even when there are no fixed coils
        root.find("group_fixed_index").text = " ".join(fixed_groups) or "-1"

        return rootET.tostring(root, encoding="unicode")
```